pkg/graveler/graveler.go

```diff
@@ -2075,6 +2078,18 @@ func (g *Graveler) Commit(ctx context.Context, repository *RepositoryRecord, bra
 	return newCommitID, nil
 }
 
+func (g *Graveler) CreateCommitRecord(ctx context.Context, repository *RepositoryRecord, commit Commit, opts ...SetOptionsFunc) (CommitID, error) {
+	options := &SetOptions{}
+	for _, opt := range opts {
+		opt(options)
+	}
+	if repository.ReadOnly && !options.Force {
+		return "", ErrReadOnlyRepository
+	}
+
+	return g.RefManager.AddCommit(ctx, repository, commit)
```

Comment:
This is somewhat dangerous because AddCommit recomputes the commit ID.  In future if we change the digest algorithm, lakeFS will continue to work but replication of existing installations will break.